ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -11258,38 +11260,41 @@ private Operator genPostGroupByBodyPlan(Operator curr, String dest, QB qb,
 
     if (qbp.getOrderByForClause(dest) != null) {
       genReduceSink = true;
+      runsSingleReducer = true;
       hasOrderBy = true;
     }
 
+    if (offset > 0) {
+      if (!hasOrderBy) {
+        final String error = HiveConf.StrictChecks.checkOffsetWithoutOrderBy(conf);
```

Comment:
It is a conversation starter:
If the checkOffsetWithoutOrderBy return value is an error message, then why not throw it inside the method? This is bit a C++ style error handling style imho. It creates more cyclomatic complexity just for error handling.